azkaban-common/src/main/java/azkaban/project/JdbcProjectImpl.java

```diff
@@ -968,4 +972,75 @@ public void cleanOlderProjectVersion(final int projectId, final int version)
       throw new ProjectManagerException("clean older project transaction failed", e);
     }
   }
+
+  @Override
+  public void uploadFlowFile(final int projectId, final int projectVersion, final int flowVersion,
+      final File flowFile) throws ProjectManagerException {
+    logger.info(String
+        .format(
+            "Uploading flow file %s, version %d for project %d, version %d, file length is [%d bytes]",
+            flowFile.getName(), flowVersion, projectId, projectVersion, flowFile.length()));
+
+    if (flowFile.length() > MAX_FLOW_FILE_SIZE) {
```

Comment:
What will happen when a user submits a right size zip after a too big zip failure? Will flowVersion increase one? Have you considered moving the logic at the beginning of uploading yml logic, like before launching database operations?

Reply:
FlowVersion will not increase because the big file is not uploaded to DB. 
User will first query DB to get the latest flow version (will be implemented in the next PR), and then call this method to upload new version of flow file to DB. This file length check happens before the uploading DB operation.